### PR TITLE
Add required field in creation of surveys

### PR DESCRIPTION
**Pull request by jaimeferreira11**
Add required input in template and fix redirect in to button cancel

### CODE REVIEW

File: src/app/surveys/add/view.js [hunk 1]
```diff
@@ -52,44 +53,46 @@ export default Mn.View.extend({
...
 
-    if (errors) {
-      errors.forEach(error => {
-        FlashesService.request('add', {
-          timeout: 2000,
-          type: 'warning',
-          title: error
+      if (errors) {
+        errors.forEach(error => {
+          FlashesService.request('add', {
```
Comment (rodrigojv) [inline src/app/surveys/add/view.js]: We can return from validate() method, an object such that:

```javascript
errors: [
   {field: 'title', message: 'is requried', required: true},
   {field: 'description', message: 'is requried', required: true}
]
```

And then we can do:

```javascript
errors.forEach(error =>{
  if (error.required)  this.$el.find(`#${error.field}`).addClass('error');
  // show error.msg
});
```

File: src/app/surveys/add/view.js [hunk 2]
```diff
@@ -54,64 +54,74 @@ export default Mn.View.extend({
   },
   handleSubmit(event) {
     if(utils.requiredInput($("#title"))  && utils.requiredInput($("#description"))){
-
       const button = utils.getLoadingButton(this.$el.find('#submit'));
-      event.preventDefault();
-      button.loading();
+      try {
```
Comment (rodrigojv) [inline src/app/surveys/add/view.js]: The try/catch encloses a block of code that is a bit long. So what we can do is create a new method `saveSurvey()` and surround that with a try/catch.

```javascript
handleSubmit() {
    try{
       this.saveySurvey();
   } catch ....
}
```

File: src/app/surveys/add/view.js [hunk 3]
```diff
@@ -54,64 +54,74 @@ export default Mn.View.extend({
...
+        .forEach(element => {
+          this.model.set(element.name, element.value);
+        });
 
-      this.validate = {};
-      this.validate.title = this.model.title;
-      this.validate.description = this.model.description;
-      this.validate.survey_schema = this.schema.getValue();
-      this.validate.survey_ui_schema = this.schemaUI.getValue();
+        this.validate = {};
```
Comment (rodrigojv) [inline src/app/surveys/add/view.js]: Here is better to use the object literal notation:
```javascript
let validate = {
   title: this.model.title
  description: this.model.description
....
}
```

File: src/app/surveys/add/view.js [hunk 4]
```diff
@@ -52,44 +53,46 @@ export default Mn.View.extend({
     };
   },
   handleSubmit(event) {
-    const button = utils.getLoadingButton(this.$el.find('#submit'));
-    event.preventDefault();
-    button.loading();
+    if(utils.requiredInput($("#title"))  && utils.requiredInput($("#description"))){
```
[the remark this answers is not in the document]
Comment (rodrigojv) [reply src/app/surveys/add/view.js]: See my comment below to see how we can handle such validation of required fields.

**Comment by jaimeferreira11** [comment]
Would you please check it again?

**Pull request merged**